eb_hooks.py

```diff
@@ -2,22 +2,79 @@
 # see https://docs.easybuild.io/en/latest/Hooks.html
 import os
 
+from easybuild.toolchains.mpi.openmpi import TC_CONSTANT_OPENMPI
 from easybuild.tools.build_log import EasyBuildError, print_msg
+from easybuild.tools.config import build_option, update_build_option
 from easybuild.tools.systemtools import POWER, get_cpu_architecture
 
 
+def get_eessi_envvar(eessi_envvar):
+    """Get an EESSI environment variable from the environment"""
+
+    eessi_envvar_value = os.getenv(eessi_envvar)
+    if eessi_envvar_value is None:
+        raise EasyBuildError("$%s is not defined!", eessi_envvar)
+
+    return eessi_envvar_value
+
+
+def get_rpath_override_dir(software_name):
+    # determine path to installations in software layer via $EESSI_SOFTWARE_PATH
+    eessi_software_path = get_eessi_envvar('EESSI_SOFTWARE_PATH')
+    eessi_pilot_version = get_eessi_envvar('EESSI_PILOT_VERSION')
+
+    # construct the rpath override directory
+    rpath_injection_dir = os.path.join(
+        # Make sure we are looking inside the `host_injections` directory
+        eessi_software_path.replace(eessi_pilot_version, os.path.join('host_injections', eessi_pilot_version), 1),
+        # Add the subdirectory for the specific software
+        'rpath_overrides',
+        software_name
+    )
+
+    return rpath_injection_dir
+
+
 def parse_hook(ec, *args, **kwargs):
     """Main parse hook: trigger custom functions based on software name."""
 
     # determine path to Prefix installation in compat layer via $EPREFIX
-    eprefix = os.getenv('EPREFIX')
-    if eprefix is None:
-        raise EasyBuildError("$EPREFIX is not defined!")
+    eprefix = get_eessi_envvar('EPREFIX')
 
     if ec.name in PARSE_HOOKS:
         PARSE_HOOKS[ec.name](ec, eprefix)
 
 
+def pre_ready_hook(self, *args, **kwargs):
+    """Main pre-ready hook: trigger custom functions."""
+
+    # Check if we have an MPI family in the toolchain (returns None if there is not)
+    mpi_family = self.toolchain.mpi_family()
```

Comment:
I originally had this so that it would also trigger this when OpenMPI was a direct dependency, but this approach is much cleaner and more general

Reply:
Makes sense to me!